Terminal.Gui/View/Layout/DimAuto.cs

```diff
@@ -17,25 +17,30 @@ namespace Terminal.Gui;
 /// </remarks>
 public class DimAuto : Dim
 {
-    private readonly Dim? _maximumContentDim;
-
-    private readonly Dim? _minimumContentDim;
-
-    private readonly DimAutoStyle _style;
-
-    /// <inheritdoc/>
+    /// <inheritdoc />
     public override bool Equals (object? other)
     {
+        if (ReferenceEquals (this, other))
+        {
+            return true;
+        }
+
         if (other is not DimAuto auto)
         {
             return false;
         }
 
-        return auto.MinimumContentDim == MinimumContentDim && auto.MaximumContentDim == MaximumContentDim && auto.Style == Style;
+        return EqualityComparer<Dim?>.Default.Equals (MinimumContentDim, auto.MinimumContentDim) &&
```

Comment:
This method of getting equality is non-ideal and, when I turn these into record structs, it'll be even less ideal.

You can actually achieve the whole equality thing by just making them all `record` instead of `class` for now. That comes with implicitly defined memberwise equality for all fields of the type, or you can override it if you _need_ to... But the implicit equality already works like the one you wrote, while also bringing along all the other goodies records bring along for free, including decompose, non-destructive mutation (the `with` operator), and built-in support of the `IEqualityOperators<TSelf,TSelf,bool>` interface.

I have like 3 open branches in which I've turned the Dim types at least into `record` and one or MAYBE two I turned them into `record struct`, but they were before the work you did to make Dim better than it was got started, so they're all pretty much useless now.

But the change to just plain `record` is really straightforward.

And if you switch it to using a primary constructor at the same time (which is optional), it will also be immutable by default, which it already is since the fields are `readonly`.

Basically, like.... any code in this file that has to do with equality just gets to go away, all by writing `record` instead of class.

`Dim` also has to be a record for that, because records are not allowed to inherit from classes and vice versa...which is weird if you ask me, because they're just classes with extra goodies written by Roslyn, and even look like normal classes to other languages anyway. 🤷‍♂️

Reply:
Oh also... Being a record means you do not need to have tests for equality, since it's handled in the framework and not in our explicit code any more.

If we, at some point, were to override equality to not mean value equality, then we'd need our own tests again. But these types are values and use value equality, so that shouldn't be an issue.